src/dispatcher.rs

```diff
@@ -0,0 +1,378 @@
+            // the _routes feature).
+            tracing::trace!(path = ?r.route_pattern, uri_fragment, "Trying route path");
+            if r.is_match(uri_fragment) {
+                return Ok(r.clone());
+            }
+        }
+
+        Err(anyhow::anyhow!("No handler for path {}", uri_fragment))
+    }
+}
+
+const DEFAULT_ENTRYPOINT: &str = "_start";
+
+impl RoutingTableEntry {
+    pub fn is_match(&self, uri_fragment: &str) -> bool {
+        self.route_pattern.is_match(uri_fragment)
+    }
+
+    fn build_from_modules_toml(source: &Loaded<ModuleMapConfigurationEntry>) -> anyhow::Result<RoutingTableEntry> {
+        let route_pattern = RoutePattern::parse(&source.metadata.route);
+        
+        let wasm_source = WasmModuleSource::Blob(source.content.clone());
+        let wasm_route_handler = WasmRouteHandler {
+            wasm_module_source: wasm_source,
+            wasm_module_name: source.metadata.module.clone(),
+            entrypoint: source.metadata.entrypoint.clone().unwrap_or_else(|| DEFAULT_ENTRYPOINT.to_owned()),
+            volumes: source.metadata.volumes.clone().unwrap_or_default(),
+            allowed_hosts: source.metadata.allowed_hosts.clone(),
+            http_max_concurrency: source.metadata.http_max_concurrency,
+        };
+        let handler_info = RouteHandler::Wasm(wasm_route_handler);
+
+        Ok(Self {
+            route_pattern,
+            handler_info,
+        })
+    }
+
+    fn build_from_bindle_entry(source: &(WagiHandlerInfo, Bits)) -> Option<anyhow::Result<RoutingTableEntry>> {
+        let (wagi_handler, bits) = source;
+
+        let route_pattern = RoutePattern::parse(&wagi_handler.route);
+        let wasm_source = WasmModuleSource::Blob(bits.wasm_module.clone());
+        let wasm_route_handler = WasmRouteHandler {
+            wasm_module_source: wasm_source,
+            wasm_module_name: wagi_handler.parcel.label.name.clone(),
+            entrypoint: wagi_handler.entrypoint.clone().unwrap_or_else(|| DEFAULT_ENTRYPOINT.to_owned()),
+            volumes: bits.volume_mounts.clone(),
+            allowed_hosts: wagi_handler.allowed_hosts.clone(),
+            http_max_concurrency: None,
+        };
+        let handler_info = RouteHandler::Wasm(wasm_route_handler);
+
+        Some(Ok(Self {
+            route_pattern,
+            handler_info,
+        }))
+    }
+
+    fn inbuilt(path: &str, handler: RouteHandler) -> Self {
+        Self {
+            route_pattern: RoutePattern::Exact(path.to_owned()),
+            handler_info: handler,
+        }
+    }
+
+    /// Returns a unique ID for the routing table entry.
+    ///
+    /// This is the SHA256 sum of the following data, written into the hasher in the following order
+    /// (skipping any `None`s):
+    ///
+    /// - route
+    /// - host
+    //
+    // TODO: this ^^ is the original comment on Module::id() but it doesn't seem to
```

Comment:
You are correct. I removed the host part when I removed virtual hosting a few months ago. Missed the comment text.